src/library/Collections.nim

```diff
@@ -76,56 +76,55 @@ proc defineSymbols*() =
             # TODO(Collections\append) Append for arrays vs strings seems to be much slower
             #  https://github.com/arturo-lang/benchmarks/blob/main/results/17-8-2022/micro.md
             #  labels: enhancement, library, performance
-            if x.kind==Literal:
-                if InPlace.kind==String:
-                    if y.kind==String:
+            if x.kind == Literal:
+                if InPlace.kind == String:
+                    if y.kind == String:
                         InPlaced.s &= y.s
-                    elif y.kind==Char:
+                    elif y.kind == Char:
                         InPlaced.s &= $(y.c)
-                elif InPlaced.kind==Char:
-                    if y.kind==String:
+                elif InPlaced.kind == Char:
+                    if y.kind == String:
                         SetInPlace(newString($(InPlaced.c) & y.s))
-                    elif y.kind==Char:
+                    elif y.kind == Char:
                         SetInPlace(newString($(InPlaced.c) & $(y.c)))
                 else:
-                    if y.kind==Block:
+                    if y.kind == Block:
                         ensureCleaned(y)
-                        for item in cleanY:
-                            InPlaced.a.add(item)
+                        InPlaced = newBlock(cleanAppend(InPlaced.a, y.a))
```

Comment:
```suggestion
                        # TODO(Collections\append) In-place appending should actually work in-place
                        #  labels: enhancement, library
                        InPlaced.a = cleanAppend(InPlaced.a, y.a)
```

`cleanAppend`, the way it's defined right now, takes two (not necessarily clean) ValueArray objects and returns a ValueArray. So, we can directly assign the result back to the `.a` field, which holds the ValueArray.

I'm adding the TODO since in-place methods are meant to work on the value itself.

Basically, here we are taking the symbol's values (`InPlaced.a`) and the second parameter's values (`y.a`), concatenate them together, get this news array and assign it back as the initial symbols values. So, technically, there is an extra array created, while we could have just appended the second parameter's values (`y.a`) to`InPlaced.a` directly.

For that we would need a 2nd version of `cleanAppend` working directly on the first parameter, e.g.:

```nim
proc cleanAppendInPlace*(dest: var ValueArray, src: ValueArray) = 
    # append src directly to dest
```

And just do `Inplaced.a.cleanAppendInPlace(y.a)` :)